src/service/resource.js

```diff
@@ -897,9 +897,12 @@ export class Resource {
 
     if (this.state_ != ResourceState.LAYOUT_SCHEDULED) {
       const err = dev().createError(
-        'startLayout called but not LAYOUT_SCHEDULED'
+        'startLayout called but not LAYOUT_SCHEDULED',
+        'currently: ',
+        this.state_
       );
-      reportError(err, this.element);
+      err.associatedElement = this.element;
+      reportError(err);
       return Promise.reject(err);
```

Comment:
Are you confident keeping this?

Reply:
Yes. The worse case is that the element is deferred to a second pass. But as soon as it goes through the flow correctly, it will do layout.